### ISSUE TITLE
Wrong tax totals per vat and also wrong gross total

### ISSUE THREAD

**Issue by charkoulis** [josemmo/einvoicing#63]
**Scenario**
2 products with 1.96 unit price including tax 13% -> 1.73 without tax  
2 products with 0.04 unit price including tax 24% -> 0.03 without tax

```
$line->setName('Product 1')
            ->setPrice(1.73)
            ->setVatRate(13)
            ->setQuantity(2);
```

```
$line->setName('Product 2')
            ->setPrice(0.03)
            ->setVatRate(24)
            ->setQuantity(2);

```

The totals should be equal to 4 but it is calculated as 3.98.
It seems that it loses 0.01 per tax rate because the rounding is being applied at the total and not on each unit price separately.

![image](https://github.com/user-attachments/assets/ebd7cd43-77a5-4ae7-837a-38a22e67903c)

**Comment by josemmo** [comment]
Hi, @Charkoulis!

For better or worse, that's how rounding works in the specification.

To account for these issues, you can make use of the rounding amount field (BT-114):
```php
// [...]
$invoice->setRoundingAmount(0.02);
```

**Comment by Charkoulis** [comment]
Hi @josemmo 

Even if i fix the total amount with the setRoundingAmount function, the taxAmount  for each VAT  will still be wrong.

**Comment by downey-lv** [comment]
Yeah, it's annoying, but the specification requires that each entry is rounded to two decimals. Currently dealing with this too, but the library is doing what it's supposed to according to the spec.

**Comment by josemmo** [comment]
Closing this issue as there's little we can do without changing the specification.
